gravitee-am-plugins-handlers/gravitee-am-plugins-handlers-botdetection/src/main/java/io/gravitee/am/plugins/botdetection/core/BotDetectionPluginManager.java

```diff
@@ -13,27 +13,33 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package io.gravitee.am.plugins.botdetection.core;
 
+import io.gravitee.am.botdetection.api.BotDetection;
+import io.gravitee.am.botdetection.api.BotDetectionConfiguration;
 import io.gravitee.am.botdetection.api.BotDetectionProvider;
-import io.gravitee.plugin.core.api.Plugin;
-
-import java.io.IOException;
-import java.util.Collection;
+import io.gravitee.am.plugins.handlers.api.core.AMPluginManager;
+import io.gravitee.am.plugins.handlers.api.core.NamedBeanFactoryPostProcessor;
+import io.gravitee.am.plugins.handlers.api.core.ProviderPluginManager;
+import io.gravitee.am.plugins.handlers.api.provider.ProviderConfiguration;
+import io.gravitee.plugin.core.api.PluginContextFactory;
 
 /**
- * @author Eric LELEU (eric.leleu at graviteesource.com)
+ * @author Rémi SULTAN (remi.sultan at graviteesource.com)
  * @author GraviteeSource Team
  */
-public interface BotDetectionPluginManager {
-
-    void register(BotDetectionDefinition definition);
-
-    Collection<Plugin> getAll();
-
-    Plugin findById(String pluginId);
-
-    BotDetectionProvider create(String type, String configuration);
-
-    String getSchema(String pluginId) throws IOException;
+public abstract class BotDetectionPluginManager
```

Comment:
The abstraction here is useful for 2 things:
- Containing the `NamedBeanPostProcessor` classes  that aren't accessible to the outside
- Not having noise of all the `ProviderPluginManager` generic implementation for reading purpose

I kind of agree right now it is made for "tidying purposes"
I still believe it's good practice to work with abstractions in case we want to proxy these or have different implementations

Reply:
This was the case for instance with the IdentityProviderManager, where more methods were necessary and thus the abstraction was useful

Reply:
Why have an abstract class that will probably only ever have one child class? I think we have different ideas of what abstract classes are for.

Reply:
I agree with you. 
It's less likely to have many implementation for just one plugin. 👍 

Let's remove the business abstraction and extract the NamedBeanFactoryPostProcessor in separate classes.